pkg/agent/types/annotations.go

```diff
@@ -23,4 +23,7 @@ const (
 
 	// NodeWireGuardPublicAnnotationKey represents the key of the Node's WireGuard public key in the Annotations of the Node.
 	NodeWireGuardPublicAnnotationKey string = "node.antrea.io/wireguard-public-key"
+
+	// ExternalIPPoolAnnotationKey is the key of the Service annotation that specifies the Service's desired external IP pool.
+	ExternalIPPoolAnnotationKey string = "antrea.io/external-ip-pool"
```

Comment:
I am not very sure about this one (adding "service" - service.antrea.io/external-ip-pool or not). Let us see what @tnqn thinks.

Reply:
sure.

Reply:
Currently all such annotations have a prefix about it's category before "antrea.io", e.g. "node.antrea.io/...", "nodeportlocal.antrea.io/...", "ipam.antrea.io", though I can't tell what's the actual benefit. Maybe we could just keep them consistent?
BTW, K8s annotation has such prefix as well: https://github.com/kubernetes/kubernetes/blob/3520e8b3393fd86064802523c36af2e0b043a85d/pkg/apis/core/annotation_key_constants.go#L80

I have a question, when it supports dual-stack, will the ip pool be a list? If yes, should it be `external-ip-pools`?
We should probably unify the format of ipam annotation in the future: https://github.com/antrea-io/antrea/blob/87c5d38b8cc2299a39af6dd2f46577aff25a8971/pkg/agent/cniserver/ipam/annotations.go

Reply:
So, you recommend service.antrea.io/external-ip-pools?

Reply:
yes